src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,205 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Checks that there is no whitespace before the colon in a switch label.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NoWhitespaceBeforeCaseDefaultColon&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *   {
+ *     switch(1) {
+ *         case 1 : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *         case 2: // OK
+ *             break;
+ *         default : // violation, whitespace before ':' is not allowed here
+ *             break;
+ *     }
+ *
+ *     switch(2) {
+ *         case 2: // OK
+ *             break;
+ *         case 3, 4
+ *                  : break; // violation, ':' should not be in separate line
+ *         case 4,
+ *               5: break; // ok
+ *         default
+ *               : // violation, ':' should not be in separate line
+ *             break;
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code ws.colonInSeparateLine}
+ * </li>
+ * <li>
+ * {@code ws.preceded}
+ * </li>
+ * </ul>
+ *
+ * @since 8.42
+ */
+@StatelessCheck
+public class NoWhitespaceBeforeCaseDefaultColonCheck
+    extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "ws.preceded";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_ON_SEPARATE = "ws.colonInSeparateLine";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST parent = ast.getParent();
+        final boolean isColonOfCase = parent.getType() == TokenTypes.LITERAL_CASE;
+        final boolean isColonOfDefault = parent.getType() == TokenTypes.LITERAL_DEFAULT;
+        final boolean violation = isColonOfCase && checkWhitespaceBeforeColonOfCase(ast)
+                || isColonOfDefault && checkWhitespaceBeforeColonOfDefault(ast);
```

Comment:
If you use the below suggestion, you can just incorporate this boolean into the `else if` branch below.

Reply:
This is not clear, can you please explain in bit detail

Reply:
On second thought, disregard that comment. 

What do you think about naming `violation` something like `isWhiteSpaceBeforeColon`? I think that would read well:

```java
        if (isColonOnDifferentLine(ast, parent)) {
            log(ast, MSG_KEY_ON_SEPARATE);
        }
        else if (isWhiteSpaceBeforeColon) {
            log(ast, MSG_KEY, ast.getText());
        }
```

This could even be a method itself (`isWhiteSpaceBeforeColon()`), and simply `return  isColonOfCase && checkWhitespaceBeforeColonOfCase(ast)
                || isColonOfDefault && checkWhitespaceBeforeColonOfDefault(ast);`, giving us:
                

```java
        if (isColonOnDifferentLine(ast, parent)) {
            log(ast, MSG_KEY_ON_SEPARATE);
        }
        else if (isWhiteSpaceBeforeColon(ast)) {
            log(ast, MSG_KEY, ast.getText());
        }
```

Reply:
> What do you think about naming violation something like isWhiteSpaceBeforeColon? I think that would read well:

Totally in its favor. Done.

Reply:
Resolved.